cmd/main.go

```diff
@@ -162,58 +160,12 @@ func ProcessOutChannel(wg *sync.WaitGroup, scConfig *common.SCConfiguration) {
 		select { //nolint:gosimple
 		case d := <-scConfig.EventOutCh: // do something that is put out by QDR
 			// regular events
-			etype := d.Data.Type()
-			var event interface{}
-			var err error
-			if etype == "HW_EVENT" {
-				event, err = v1hwevent.GetCloudNativeEvents(*d.Data)
-			} else {
-				event, err = v1event.GetCloudNativeEvents(*d.Data)
-			}
-			if err != nil {
-				log.Errorf("error marshalling event data when reading from amqp %v\n %#v", err, d)
-				log.Infof("data %#v", d.Data)
-			} else if d.Type == channel.EVENT {
+			if d.Type == channel.EVENT {
 				if d.Status == channel.NEW {
-					if d.ProcessEventFn != nil { // always leave event to handle by default method for events
-						if err := d.ProcessEventFn(event); err != nil {
-							log.Errorf("error processing data %v", err)
-							localmetrics.UpdateEventReceivedCount(d.Address, localmetrics.FAILED)
-						}
-					} else if sub, ok := scConfig.PubSubAPI.HasSubscription(d.Address); ok {
-						if sub.EndPointURI != nil {
-							if etype == "HW_EVENT" {
-								event, ok := event.(hwevent.Event)
-								if !ok {
-									log.Errorf("got data of type %T but wanted hwevent.Event", event)
-								}
-								event.ID = sub.ID // set ID to the subscriptionID
-								if err := restClient.PostHwEvent(sub.EndPointURI, event); err != nil {
-									log.Errorf("error posting request at %s", sub.EndPointURI)
-									localmetrics.UpdateEventReceivedCount(d.Address, localmetrics.FAILED)
-								} else {
-									localmetrics.UpdateEventReceivedCount(d.Address, localmetrics.SUCCESS)
-								}
-							} else {
-								event, ok := event.(ceevent.Event)
-								if !ok {
-									log.Errorf("got data of type %T but wanted ceevent.Event", event)
-								}
-								event.ID = sub.ID // set ID to the subscriptionID
-								if err := restClient.PostEvent(sub.EndPointURI, event); err != nil {
-									log.Errorf("error posting request at %s", sub.EndPointURI)
-									localmetrics.UpdateEventReceivedCount(d.Address, localmetrics.FAILED)
-								} else {
-									localmetrics.UpdateEventReceivedCount(d.Address, localmetrics.SUCCESS)
-								}
-							}
-						} else {
-							log.Warnf("endpoint uri not given, posting event to log %#v for address %s\n", event, d.Address)
-							localmetrics.UpdateEventReceivedCount(d.Address, localmetrics.SUCCESS)
-						}
+					if d.Data.Type() == "HW_EVENT" {
```

Comment:
This is the free text for Now.  I will redesign the event pipeline to accommodate eventSubType soon,